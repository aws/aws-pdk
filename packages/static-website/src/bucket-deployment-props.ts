// ~~ Generated by projen. To modify, edit .projenrc.js and run "npx projen".
import { aws_cloudfront, aws_ec2, aws_iam, aws_logs, aws_s3, aws_s3_deployment, Expiration, Size } from 'aws-cdk-lib';

/**
 * BucketDeploymentProps
 */
export interface BucketDeploymentProps {
  /**
   * System-defined x-amz-website-redirect-location metadata to be set on all objects in the deployment.
   * @default - No website redirection.
   * @stability stable
   */
  readonly websiteRedirectLocation?: string;
  /**
   * Where in the VPC to place the deployment lambda handler.
   * Only used if 'vpc' is supplied.
   * @default - the Vpc default strategy if not specified
   * @stability stable
   */
  readonly vpcSubnets?: aws_ec2.SubnetSelection;
  /**
   * The VPC network to place the deployment lambda handler in.
   * This is required if `useEfs` is set.
   * @default None
   * @stability stable
   */
  readonly vpc?: aws_ec2.IVpc;
  /**
   * Mount an EFS file system.
   * Enable this if your assets are large and you encounter disk space errors.
   * Enabling this option will require a VPC to be specified.
   * @default - No EFS. Lambda has access only to 512MB of disk space.
   * @stability stable
   */
  readonly useEfs?: boolean;
  /**
   * System-defined x-amz-storage-class metadata to be set on all objects in the deployment.
   * @default - Default storage-class for the bucket is used.
   * @stability stable
   */
  readonly storageClass?: aws_s3_deployment.StorageClass;
  /**
   * If set to true, uploads will precompute the value of `x-amz-content-sha256` and include it in the signed S3 request headers.
   * @default - `x-amz-content-sha256` will not be computed
   * @stability stable
   */
  readonly signContent?: boolean;
  /**
   * System-defined x-amz-server-side-encryption-customer-algorithm metadata to be set on all objects in the deployment.
   * Warning: This is not a useful parameter until this bug is fixed: https://github.com/aws/aws-cdk/issues/6080
   * @default - Not set.
   * @stability stable
   */
  readonly serverSideEncryptionCustomerAlgorithm?: string;
  /**
   * System-defined x-amz-server-side-encryption-aws-kms-key-id metadata to be set on all objects in the deployment.
   * @default - Not set.
   * @stability stable
   */
  readonly serverSideEncryptionAwsKmsKeyId?: string;
  /**
   * System-defined x-amz-server-side-encryption metadata to be set on all objects in the deployment.
   * @default - Server side encryption is not used.
   * @stability stable
   */
  readonly serverSideEncryption?: aws_s3_deployment.ServerSideEncryption;
  /**
   * Execution role associated with this function.
   * @default - A role is automatically created
   * @stability stable
   */
  readonly role?: aws_iam.IRole;
  /**
   * If this is set to "false", the destination files will be deleted when the resource is deleted or the destination is updated.
   * NOTICE: Configuring this to "false" might have operational implications. Please
   * visit to the package documentation referred below to make sure you fully understand those implications.
   * @default true - when resource is deleted/updated, files are retained
   * @stability stable
   */
  readonly retainOnDelete?: boolean;
  /**
   * If this is set to false, files in the destination bucket that do not exist in the asset, will NOT be deleted during deployment (create/update).
   * @default true
   * @stability stable
   */
  readonly prune?: boolean;
  /**
   * User-defined object metadata to be set on all objects in the deployment.
   * @default - No user metadata is set
   * @stability stable
   */
  readonly metadata?: Record<string, string>;
  /**
   * The amount of memory (in MiB) to allocate to the AWS Lambda function which replicates the files from the CDK bucket to the destination bucket.
   * If you are deploying large files, you will need to increase this number
   * accordingly.
   * @default 128
   * @stability stable
   */
  readonly memoryLimit?: number;
  /**
   * The number of days that the lambda function's log events are kept in CloudWatch Logs.
   * @default logs.RetentionDays.INFINITE
   * @stability stable
   */
  readonly logRetention?: aws_logs.RetentionDays;
  /**
   * If this is set, matching files or objects will be included with the deployment's sync command.
   * Since all files from the deployment package are included by default, this property
   * is usually leveraged alongside an `exclude` filter.
   * @default - No include filters are used and all files are included with the sync command
   * @stability stable
   */
  readonly include?: Array<string>;
  /**
   * If this is set, the zip file will be synced to the destination S3 bucket and extracted.
   * If false, the file will remain zipped in the destination bucket.
   * @default true
   * @stability stable
   */
  readonly extract?: boolean;
  /**
   * System-defined expires metadata to be set on all objects in the deployment.
   * @default - The objects in the distribution will not expire.
   * @stability stable
   */
  readonly expires?: Expiration;
  /**
   * If this is set, matching files or objects will be excluded from the deployment's sync command.
   * This can be used to exclude a file from being pruned in the destination bucket.
   *
   * If you want to just exclude files from the deployment package (which excludes these files
   * evaluated when invalidating the asset), you should leverage the `exclude` property of
   * `AssetOptions` when defining your source.
   * @default - No exclude filters are used
   * @stability stable
   */
  readonly exclude?: Array<string>;
  /**
   * The size of the AWS Lambda functionâ€™s /tmp directory in MiB.
   * @default 512 MiB
   * @stability stable
   */
  readonly ephemeralStorageSize?: Size;
  /**
   * The file paths to invalidate in the CloudFront distribution.
   * @default - All files under the destination bucket key prefix will be invalidated.
   * @stability stable
   */
  readonly distributionPaths?: Array<string>;
  /**
   * The CloudFront distribution using the destination bucket as an origin.
   * Files in the distribution's edge caches will be invalidated after
   * files are uploaded to the destination bucket.
   * @default - No invalidation occurs
   * @stability stable
   */
  readonly distribution?: aws_cloudfront.IDistribution;
  /**
   * Key prefix in the destination bucket.
   * Must be <=104 characters
   * @default "/" (unzip to root of the destination bucket)
   * @stability stable
   */
  readonly destinationKeyPrefix?: string;
  /**
   * System-defined content-type metadata to be set on all objects in the deployment.
   * @default - Not set.
   * @stability stable
   */
  readonly contentType?: string;
  /**
   * System-defined content-language metadata to be set on all objects in the deployment.
   * @default - Not set.
   * @stability stable
   */
  readonly contentLanguage?: string;
  /**
   * System-defined content-encoding metadata to be set on all objects in the deployment.
   * @default - Not set.
   * @stability stable
   */
  readonly contentEncoding?: string;
  /**
   * System-defined cache-disposition metadata to be set on all objects in the deployment.
   * @default - Not set.
   * @stability stable
   */
  readonly contentDisposition?: string;
  /**
   * System-defined cache-control metadata to be set on all objects in the deployment.
   * @default - Not set.
   * @stability stable
   */
  readonly cacheControl?: Array<aws_s3_deployment.CacheControl>;
  /**
   * System-defined x-amz-acl metadata to be set on all objects in the deployment.
   * @default - Not set.
   * @stability stable
   */
  readonly accessControl?: aws_s3.BucketAccessControl;
  /**
   * The sources from which to deploy the contents of this bucket.
   * @stability stable
   */
  readonly sources?: Array<aws_s3_deployment.ISource>;
  /**
   * The S3 bucket to sync the contents of the zip file to.
   * @stability stable
   */
  readonly destinationBucket?: aws_s3.IBucket;
}
