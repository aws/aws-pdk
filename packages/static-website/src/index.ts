/*! Copyright [Amazon.com](http://amazon.com/), Inc. or its affiliates. All Rights Reserved.
SPDX-License-Identifier: Apache-2.0 */
export * from "./cloudfront-web-acl";
export * from "./static-website";
export * from "./bucket-deployment-props";
export * from "./distribution-props";
