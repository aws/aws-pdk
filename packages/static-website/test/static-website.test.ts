/*! Copyright [Amazon.com](http://amazon.com/), Inc. or its affiliates. All Rights Reserved.
SPDX-License-Identifier: Apache-2.0 */
import path from "path";
import { PDKNag } from "@aws-prototyping-sdk/pdk-nag";
import { NestedStack, Stack } from "aws-cdk-lib";
import { Template } from "aws-cdk-lib/assertions";
import { StaticWebsite } from "../src";

describe("Static Website Unit Tests", () => {
  it("Defaults", () => {
    const stack = new Stack(PDKNag.app());
    new StaticWebsite(stack, "Defaults", {
      websiteContentPath: path.join(__dirname, "./sample-website"),
    });

    expect(Template.fromStack(stack)).toMatchSnapshot();
  });

  it("Defaults - Nested", () => {
    const stack = new Stack(PDKNag.app());
    const nestedStack = new NestedStack(stack, "Nested-Stack");
    new StaticWebsite(nestedStack, "Defaults", {
      websiteContentPath: path.join(__dirname, "./sample-website"),
    });

    expect(Template.fromStack(nestedStack)).toMatchSnapshot();
  });
});
