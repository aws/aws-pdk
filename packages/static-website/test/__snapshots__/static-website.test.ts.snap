// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Static Website Unit Tests Defaults 1`] = `
Object {
  "Parameters": Object {
    "BootstrapVersion": Object {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": Object {
    "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756C81C01536": Object {
      "DependsOn": Array [
        "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756CServiceRoleDefaultPolicy88902FDF",
        "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756CServiceRole89A01265",
      ],
      "Properties": Object {
        "Code": Object {
          "S3Bucket": Object {
            "Fn::Sub": "cdk-hnb659fds-assets-\${AWS::AccountId}-\${AWS::Region}",
          },
          "S3Key": "f98b78092dcdd31f5e6d47489beb5f804d4835ef86a8085d0a2053cb9ae711da.zip",
        },
        "Handler": "index.handler",
        "Layers": Array [
          Object {
            "Ref": "DefaultsWebsiteDeploymentAwsCliLayerD5AA12CC",
          },
        ],
        "Role": Object {
          "Fn::GetAtt": Array [
            "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756CServiceRole89A01265",
            "Arn",
          ],
        },
        "Runtime": "python3.7",
        "Timeout": 900,
      },
      "Type": "AWS::Lambda::Function",
    },
    "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756CServiceRole89A01265": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756CServiceRoleDefaultPolicy88902FDF": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "s3:GetObject*",
                "s3:GetBucket*",
                "s3:List*",
              ],
              "Effect": "Allow",
              "Resource": Array [
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      "arn:",
                      Object {
                        "Ref": "AWS::Partition",
                      },
                      ":s3:::",
                      Object {
                        "Fn::Sub": "cdk-hnb659fds-assets-\${AWS::AccountId}-\${AWS::Region}",
                      },
                    ],
                  ],
                },
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      "arn:",
                      Object {
                        "Ref": "AWS::Partition",
                      },
                      ":s3:::",
                      Object {
                        "Fn::Sub": "cdk-hnb659fds-assets-\${AWS::AccountId}-\${AWS::Region}",
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
            Object {
              "Action": Array [
                "s3:GetObject*",
                "s3:GetBucket*",
                "s3:List*",
                "s3:DeleteObject*",
                "s3:PutObject",
                "s3:PutObjectLegalHold",
                "s3:PutObjectRetention",
                "s3:PutObjectTagging",
                "s3:PutObjectVersionTagging",
                "s3:Abort*",
              ],
              "Effect": "Allow",
              "Resource": Array [
                Object {
                  "Fn::GetAtt": Array [
                    "DefaultsWebsiteBucket3263D025",
                    "Arn",
                  ],
                },
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      Object {
                        "Fn::GetAtt": Array [
                          "DefaultsWebsiteBucket3263D025",
                          "Arn",
                        ],
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
            Object {
              "Action": Array [
                "kms:Decrypt",
                "kms:DescribeKey",
                "kms:Encrypt",
                "kms:ReEncrypt*",
                "kms:GenerateDataKey*",
              ],
              "Effect": "Allow",
              "Resource": Object {
                "Fn::GetAtt": Array [
                  "DefaultsWebsiteBucketKeyAD72391B",
                  "Arn",
                ],
              },
            },
            Object {
              "Action": Array [
                "cloudfront:GetInvalidation",
                "cloudfront:CreateInvalidation",
              ],
              "Effect": "Allow",
              "Resource": "*",
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756CServiceRoleDefaultPolicy88902FDF",
        "Roles": Array [
          Object {
            "Ref": "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756CServiceRole89A01265",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "CustomS3AutoDeleteObjectsCustomResourceProviderHandler9D90184F": Object {
      "DependsOn": Array [
        "CustomS3AutoDeleteObjectsCustomResourceProviderRole3B1BD092",
      ],
      "Properties": Object {
        "Code": Object {
          "S3Bucket": Object {
            "Fn::Sub": "cdk-hnb659fds-assets-\${AWS::AccountId}-\${AWS::Region}",
          },
          "S3Key": "0758db37a61ce7177f50559cad0340df233a2001b5880615afa517203dd67c21.zip",
        },
        "Description": Object {
          "Fn::Join": Array [
            "",
            Array [
              "Lambda function for auto-deleting objects in ",
              Object {
                "Ref": "DefaultsWebsiteBucket3263D025",
              },
              " S3 bucket.",
            ],
          ],
        },
        "Handler": "__entrypoint__.handler",
        "MemorySize": 128,
        "Role": Object {
          "Fn::GetAtt": Array [
            "CustomS3AutoDeleteObjectsCustomResourceProviderRole3B1BD092",
            "Arn",
          ],
        },
        "Runtime": "nodejs12.x",
        "Timeout": 900,
      },
      "Type": "AWS::Lambda::Function",
    },
    "CustomS3AutoDeleteObjectsCustomResourceProviderRole3B1BD092": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Sub": "arn:\${AWS::Partition}:iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "DefaultsCloudfrontDistributionF4EA1054": Object {
      "Properties": Object {
        "DistributionConfig": Object {
          "CustomErrorResponses": Array [
            Object {
              "ErrorCode": 404,
              "ResponseCode": 200,
              "ResponsePagePath": "/index.html",
            },
          ],
          "DefaultCacheBehavior": Object {
            "CachePolicyId": "658327ea-f89d-4fab-a63d-7e88639e58f6",
            "Compress": true,
            "TargetOriginId": "DefaultsCloudfrontDistributionOrigin1BA23CD94",
            "ViewerProtocolPolicy": "redirect-to-https",
          },
          "DefaultRootObject": "index.html",
          "Enabled": true,
          "HttpVersion": "http2",
          "IPV6Enabled": true,
          "Origins": Array [
            Object {
              "DomainName": Object {
                "Fn::GetAtt": Array [
                  "DefaultsWebsiteBucket3263D025",
                  "RegionalDomainName",
                ],
              },
              "Id": "DefaultsCloudfrontDistributionOrigin1BA23CD94",
              "S3OriginConfig": Object {
                "OriginAccessIdentity": Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      "origin-access-identity/cloudfront/",
                      Object {
                        "Ref": "DefaultsCloudfrontDistributionOrigin1S3OriginCFA7596E",
                      },
                    ],
                  ],
                },
              },
            },
          ],
          "WebACLId": Object {
            "Fn::GetAtt": Array [
              "DefaultsWebsiteAclCFWebAclCustomResourceB050DB2C",
              "WebAclArn",
            ],
          },
        },
      },
      "Type": "AWS::CloudFront::Distribution",
    },
    "DefaultsCloudfrontDistributionOrigin1S3OriginCFA7596E": Object {
      "Properties": Object {
        "CloudFrontOriginAccessIdentityConfig": Object {
          "Comment": "Identity for DefaultsCloudfrontDistributionOrigin1BA23CD94",
        },
      },
      "Type": "AWS::CloudFront::CloudFrontOriginAccessIdentity",
    },
    "DefaultsWebsiteAclCFWebAclCustomResourceB050DB2C": Object {
      "DeletionPolicy": "Delete",
      "Properties": Object {
        "ID": "WebsiteAcl",
        "MANAGED_RULES": Array [
          Object {
            "name": "AWSManagedRulesCommonRuleSet",
            "vendor": "AWS",
          },
        ],
        "ServiceToken": Object {
          "Fn::GetAtt": Array [
            "DefaultsWebsiteAclCloudfrontWebAclProviderframeworkonEvent595963CB",
            "Arn",
          ],
        },
      },
      "Type": "AWS::CloudFormation::CustomResource",
      "UpdateReplacePolicy": "Delete",
    },
    "DefaultsWebsiteAclCloudfrontWebAclOnEventHandlerEC085AE0": Object {
      "DependsOn": Array [
        "DefaultsWebsiteAclCloudfrontWebAclOnEventHandlerServiceRoleDefaultPolicy916A8D60",
        "DefaultsWebsiteAclCloudfrontWebAclOnEventHandlerServiceRole55188EEB",
      ],
      "Properties": Object {
        "Code": Object {
          "S3Bucket": Object {
            "Fn::Sub": "cdk-hnb659fds-assets-\${AWS::AccountId}-\${AWS::Region}",
          },
          "S3Key": "a6f271fc89215e4bb8ddfd8ac9e56a0666ee4cf19ccc63ce69781ed196d378d5.zip",
        },
        "Environment": Object {
          "Variables": Object {
            "AWS_NODEJS_CONNECTION_REUSE_ENABLED": "1",
          },
        },
        "FunctionName": "CloudfrontWebAclCustomResource",
        "Handler": "index.onEvent",
        "Role": Object {
          "Fn::GetAtt": Array [
            "DefaultsWebsiteAclCloudfrontWebAclOnEventHandlerServiceRole55188EEB",
            "Arn",
          ],
        },
        "Runtime": "nodejs16.x",
        "Timeout": 300,
      },
      "Type": "AWS::Lambda::Function",
    },
    "DefaultsWebsiteAclCloudfrontWebAclOnEventHandlerServiceRole55188EEB": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "DefaultsWebsiteAclCloudfrontWebAclOnEventHandlerServiceRoleDefaultPolicy916A8D60": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "wafv2:CreateWebACL",
                "wafv2:DeleteWebACL",
                "wafv2:UpdateWebACL",
                "wafv2:GetWebACL",
                "wafv2:CreateIPSet",
                "wafv2:DeleteIPSet",
                "wafv2:UpdateIPSet",
                "wafv2:GetIPSet",
              ],
              "Effect": "Allow",
              "Resource": "*",
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "DefaultsWebsiteAclCloudfrontWebAclOnEventHandlerServiceRoleDefaultPolicy916A8D60",
        "Roles": Array [
          Object {
            "Ref": "DefaultsWebsiteAclCloudfrontWebAclOnEventHandlerServiceRole55188EEB",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "DefaultsWebsiteAclCloudfrontWebAclProviderframeworkonEvent595963CB": Object {
      "DependsOn": Array [
        "DefaultsWebsiteAclCloudfrontWebAclProviderframeworkonEventServiceRoleDefaultPolicyEEB18523",
        "DefaultsWebsiteAclCloudfrontWebAclProviderframeworkonEventServiceRole439E39E5",
      ],
      "Properties": Object {
        "Code": Object {
          "S3Bucket": Object {
            "Fn::Sub": "cdk-hnb659fds-assets-\${AWS::AccountId}-\${AWS::Region}",
          },
          "S3Key": "eaeb69bc290b516fe3b049f89d6118b22249df682fbabf56af300cf345198574.zip",
        },
        "Description": "AWS CDK resource provider framework - onEvent (Default/Defaults/WebsiteAcl/CloudfrontWebAclProvider)",
        "Environment": Object {
          "Variables": Object {
            "USER_ON_EVENT_FUNCTION_ARN": Object {
              "Fn::GetAtt": Array [
                "DefaultsWebsiteAclCloudfrontWebAclOnEventHandlerEC085AE0",
                "Arn",
              ],
            },
          },
        },
        "Handler": "framework.onEvent",
        "Role": Object {
          "Fn::GetAtt": Array [
            "DefaultsWebsiteAclCloudfrontWebAclProviderframeworkonEventServiceRole439E39E5",
            "Arn",
          ],
        },
        "Runtime": "nodejs12.x",
        "Timeout": 900,
      },
      "Type": "AWS::Lambda::Function",
    },
    "DefaultsWebsiteAclCloudfrontWebAclProviderframeworkonEventServiceRole439E39E5": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "DefaultsWebsiteAclCloudfrontWebAclProviderframeworkonEventServiceRoleDefaultPolicyEEB18523": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "lambda:InvokeFunction",
              "Effect": "Allow",
              "Resource": Array [
                Object {
                  "Fn::GetAtt": Array [
                    "DefaultsWebsiteAclCloudfrontWebAclOnEventHandlerEC085AE0",
                    "Arn",
                  ],
                },
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      Object {
                        "Fn::GetAtt": Array [
                          "DefaultsWebsiteAclCloudfrontWebAclOnEventHandlerEC085AE0",
                          "Arn",
                        ],
                      },
                      ":*",
                    ],
                  ],
                },
              ],
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "DefaultsWebsiteAclCloudfrontWebAclProviderframeworkonEventServiceRoleDefaultPolicyEEB18523",
        "Roles": Array [
          Object {
            "Ref": "DefaultsWebsiteAclCloudfrontWebAclProviderframeworkonEventServiceRole439E39E5",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "DefaultsWebsiteBucket3263D025": Object {
      "DeletionPolicy": "Delete",
      "Properties": Object {
        "BucketEncryption": Object {
          "ServerSideEncryptionConfiguration": Array [
            Object {
              "ServerSideEncryptionByDefault": Object {
                "KMSMasterKeyID": Object {
                  "Fn::GetAtt": Array [
                    "DefaultsWebsiteBucketKeyAD72391B",
                    "Arn",
                  ],
                },
                "SSEAlgorithm": "aws:kms",
              },
            },
          ],
        },
        "LoggingConfiguration": Object {
          "LogFilePrefix": "access-logs",
        },
        "Tags": Array [
          Object {
            "Key": "aws-cdk:auto-delete-objects",
            "Value": "true",
          },
          Object {
            "Key": "aws-cdk:cr-owned:fb67e0d5",
            "Value": "true",
          },
        ],
        "VersioningConfiguration": Object {
          "Status": "Enabled",
        },
      },
      "Type": "AWS::S3::Bucket",
      "UpdateReplacePolicy": "Delete",
    },
    "DefaultsWebsiteBucketAutoDeleteObjectsCustomResourceD840F8F2": Object {
      "DeletionPolicy": "Delete",
      "DependsOn": Array [
        "DefaultsWebsiteBucketPolicy594E6643",
      ],
      "Properties": Object {
        "BucketName": Object {
          "Ref": "DefaultsWebsiteBucket3263D025",
        },
        "ServiceToken": Object {
          "Fn::GetAtt": Array [
            "CustomS3AutoDeleteObjectsCustomResourceProviderHandler9D90184F",
            "Arn",
          ],
        },
      },
      "Type": "Custom::S3AutoDeleteObjects",
      "UpdateReplacePolicy": "Delete",
    },
    "DefaultsWebsiteBucketKeyAD72391B": Object {
      "DeletionPolicy": "Retain",
      "Properties": Object {
        "Description": "Created by Default/Defaults/WebsiteBucket",
        "KeyPolicy": Object {
          "Statement": Array [
            Object {
              "Action": "kms:*",
              "Effect": "Allow",
              "Principal": Object {
                "AWS": Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      "arn:",
                      Object {
                        "Ref": "AWS::Partition",
                      },
                      ":iam::",
                      Object {
                        "Ref": "AWS::AccountId",
                      },
                      ":root",
                    ],
                  ],
                },
              },
              "Resource": "*",
            },
          ],
          "Version": "2012-10-17",
        },
        "Tags": Array [
          Object {
            "Key": "aws-cdk:cr-owned:fb67e0d5",
            "Value": "true",
          },
        ],
      },
      "Type": "AWS::KMS::Key",
      "UpdateReplacePolicy": "Retain",
    },
    "DefaultsWebsiteBucketPolicy594E6643": Object {
      "Properties": Object {
        "Bucket": Object {
          "Ref": "DefaultsWebsiteBucket3263D025",
        },
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "s3:*",
              "Condition": Object {
                "Bool": Object {
                  "aws:SecureTransport": "false",
                },
              },
              "Effect": "Deny",
              "Principal": Object {
                "AWS": "*",
              },
              "Resource": Array [
                Object {
                  "Fn::GetAtt": Array [
                    "DefaultsWebsiteBucket3263D025",
                    "Arn",
                  ],
                },
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      Object {
                        "Fn::GetAtt": Array [
                          "DefaultsWebsiteBucket3263D025",
                          "Arn",
                        ],
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
            Object {
              "Action": Array [
                "s3:GetBucket*",
                "s3:List*",
                "s3:DeleteObject*",
              ],
              "Effect": "Allow",
              "Principal": Object {
                "AWS": Object {
                  "Fn::GetAtt": Array [
                    "CustomS3AutoDeleteObjectsCustomResourceProviderRole3B1BD092",
                    "Arn",
                  ],
                },
              },
              "Resource": Array [
                Object {
                  "Fn::GetAtt": Array [
                    "DefaultsWebsiteBucket3263D025",
                    "Arn",
                  ],
                },
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      Object {
                        "Fn::GetAtt": Array [
                          "DefaultsWebsiteBucket3263D025",
                          "Arn",
                        ],
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
            Object {
              "Action": "s3:GetObject",
              "Effect": "Allow",
              "Principal": Object {
                "CanonicalUser": Object {
                  "Fn::GetAtt": Array [
                    "DefaultsCloudfrontDistributionOrigin1S3OriginCFA7596E",
                    "S3CanonicalUserId",
                  ],
                },
              },
              "Resource": Object {
                "Fn::Join": Array [
                  "",
                  Array [
                    Object {
                      "Fn::GetAtt": Array [
                        "DefaultsWebsiteBucket3263D025",
                        "Arn",
                      ],
                    },
                    "/*",
                  ],
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::S3::BucketPolicy",
    },
    "DefaultsWebsiteDeploymentAwsCliLayerD5AA12CC": Object {
      "Properties": Object {
        "Content": Object {
          "S3Bucket": Object {
            "Fn::Sub": "cdk-hnb659fds-assets-\${AWS::AccountId}-\${AWS::Region}",
          },
          "S3Key": "8ad7bbf8be94e05d569da95ddb82511dcc959f25054825394cbb86028ccd1b6a.zip",
        },
        "Description": "/opt/awscli/aws",
      },
      "Type": "AWS::Lambda::LayerVersion",
    },
    "DefaultsWebsiteDeploymentCustomResource326CD6C1": Object {
      "DeletionPolicy": "Delete",
      "Properties": Object {
        "DestinationBucketName": Object {
          "Ref": "DefaultsWebsiteBucket3263D025",
        },
        "DistributionId": Object {
          "Ref": "DefaultsCloudfrontDistributionF4EA1054",
        },
        "Prune": true,
        "ServiceToken": Object {
          "Fn::GetAtt": Array [
            "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756C81C01536",
            "Arn",
          ],
        },
        "SourceBucketNames": Array [
          Object {
            "Fn::Sub": "cdk-hnb659fds-assets-\${AWS::AccountId}-\${AWS::Region}",
          },
        ],
        "SourceObjectKeys": Array [
          "a79f62b4071246acc1e8e834ba67dc3bbf15a3662e39d31667fa59315ef86f56.zip",
        ],
      },
      "Type": "Custom::CDKBucketDeployment",
      "UpdateReplacePolicy": "Delete",
    },
  },
  "Rules": Object {
    "CheckBootstrapVersion": Object {
      "Assertions": Array [
        Object {
          "Assert": Object {
            "Fn::Not": Array [
              Object {
                "Fn::Contains": Array [
                  Array [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  Object {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;
